src/internal_handlers/peer_management/tester.rs

```diff
@@ -9,28 +9,50 @@ use crate::{
     handlers::{MessageHandler, MessageHandlers},
     internal_handlers::peer_management::{announcement::Announcement, PeerInfo},
     network_manager::PeerNetManager,
+    peer::HandshakeHandler,
     peer_id::PeerId,
     transports::{endpoint::Endpoint, OutConnectionConfig, TcpOutConnectionConfig, TransportType},
 };
 
 use super::{PeerDB, SharedPeerDB};
 
+#[derive(Clone)]
+pub struct EmptyHandshake {
+    peer_db: SharedPeerDB,
+}
+
+impl HandshakeHandler for EmptyHandshake {
+    fn perform_handshake(
+        &mut self,
+        keypair: &KeyPair,
+        endpoint: &mut Endpoint,
+        _: &HashMap<SocketAddr, TransportType>,
+        _: &MessageHandlers,
+    ) -> Result<PeerId, PeerNetError> {
+        let id = PeerId::from_public_key(keypair.get_public_key());
```

Comment:
This could be removed as we fetch a peerid above

Reply:
The ID that we fetch on this line is the ID of the Peer ? So we can just skip the serialization of the peer ID in the data no ?
Why do we send it anyway in this case ?

Reply:
the deserialization you meant ?
I placed this as a placeholder before. But the handshake should return the distant peerid so have it above you can remove this line.